pulldown-cmark/src/parse.rs

```diff
@@ -225,6 +243,8 @@ impl<'input, F: BrokenLinkCallback<'input>> Parser<'input, F> {
             inline_stack,
             link_stack,
             html_scan_guard,
+            // always allow 100KiB
+            link_ref_expansion_limit: text.len().max(100_000),
```

Comment:
I guess I'd rather make this `text.len() + 100_000` but replicating cmark is good.